### PR TITLE
Add schema validation

### PR DESCRIPTION
**Pull request by HeinrichvonStein**
This PR addresses https://github.com/powersync-ja/powersync-js/issues/431#issuecomment-2522158014.

Add validation to the Schema constructor.
- We need to validate that the tables have a name here because a user could pass in an array Tables that don't have a name because they are using the V2 syntax.  Therefore, 'convertToClassicTables' won't be called on the tables resulting in a runtime error. 

Additional Tests:
- Added a new Schema test which checks that an error is thrown when a Schema is created incorrectly.

### CODE REVIEW

**Comment by HeinrichvonStein** [comment]
Converting to a draft for now due to validation oversite

File: packages/common/tests/db/schema/Schema.test.ts [hunk 1]
```diff
@@ -2,14 +2,26 @@ import { describe, it, expect } from 'vitest';
 import { Schema } from '../../../src/db/schema/Schema';
 import { Table } from '../../../src/db/schema/Table';
 import { column, ColumnType } from '../../../src/db/schema/Column';
+import { Column } from '../../../lib';
 
 describe('Schema', () => {
+  it('should fail if array is passed and not a table object', () => {
```
Comment (DominicGBauer) [inline packages/common/tests/db/schema/Schema.test.ts:8]: Can you specify that it should fail if an array of tables using the new syntax is passed so "should fail if an array of tables using the new syntax is passed to schema"

File: packages/common/tests/db/schema/Schema.test.ts [hunk 2]
```diff
@@ -2,14 +2,26 @@ import { describe, it, expect } from 'vitest';
...
+    const table1 = new Table({ name: column.text });
+    const table2 = new Table({ age: { type: ColumnType.INTEGER } });
+    expect(() => new Schema([table1, table2])).toThrow();
+  });
+
   it('should create a schema with an array of tables', () => {
```
Comment (DominicGBauer) [inline packages/common/tests/db/schema/Schema.test.ts:14]: "should create a schema with an array of tables using the old syntax"
just for clarity for someone who does not have context

**Review by DominicGBauer — DISMISSED** [review]

File: packages/common/tests/db/schema/Schema.test.ts [hunk 3]
```diff
@@ -2,14 +2,26 @@ import { describe, it, expect } from 'vitest';
 import { Schema } from '../../../src/db/schema/Schema';
 import { Table } from '../../../src/db/schema/Table';
 import { column, ColumnType } from '../../../src/db/schema/Column';
+import { Column } from '../../../lib';
```
Comment (stevensJourney) [inline packages/common/tests/db/schema/Schema.test.ts:5]: Nit: Importing from `lib` is technically okay for tests, but it's probably nicer to import from `src` for consistency.

**Review by stevensJourney — APPROVED** [review]

**Review by DominicGBauer — APPROVED** [review]

**Pull request merged**